### PR TITLE
add filter logs to pipeline

### PR DESCRIPTION
**Pull request by chaitanyaphalak**
## Proposed changes

Add filtering of logs capability

### CODE REVIEW

**Review by rockb1017 — APPROVED** [review]

**Review by rockb1017 — CHANGES_REQUESTED** [review]
CI fails. the agent is failing with this email 
```
Error: cannot load configuration: error reading processors configuration for filter/log: 1 error(s) decoding:

* '' has invalid keys: logs
```

**Comment by chaitanyaphalak** [comment]
thats because the agent running the CI dosent have the latest changes to filter log. 
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/4895 has to be merged first

File: charts/sck-otel/templates/_config.tpl [hunk 1]
```diff
@@ -293,6 +293,14 @@ processors:
     - key: service.name
       from_attribute: k8s.pod.labels.app
       action: upsert
+  filter/log:
+    logs:
+      # any logs matching filters are excluded from remainder of pipeline
+      exclude:
+        match_type: strict
+        resource_attributes:
+          - key: com.splunk.exclude
```
Comment (dmitryax) [inline charts/sck-otel/templates/_config.tpl]: Is `com.splunk.exclude` attribute set anywhere in the pipeline before? Shouldn't it be `k8s.pod.annotations.splunk.com/exclude` and `k8s.namespace.annotations.splunk.com/exclude` instead?

**Comment by rockb1017** [comment]
it has this error from agent. ```* '' has invalid keys: logs
2021/10/05 23:25:47 main.go:95: application run finished with error: cannot load configuration: error reading processors configuration for filter/namespacelogs: 1 error(s) decoding:```

When CI fails, you can check the stdout on "deploy log generator" steps because it runs `kubectl logs` command.

**Comment by rockb1017** [comment]
Could you re enable the filter test case ? 
line 90 in the `test/k8s_logging_tests/test_config_logging.py`

**Review by rockb1017 — APPROVED** [review]

**Pull request merged**